src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -47,6 +54,13 @@ public Http3Connection(Http3ConnectionContext context)
             _systemClock = context.ServiceContext.SystemClock;
             _timeoutControl = new TimeoutControl(this);
             _context.TimeoutControl ??= _timeoutControl;
+
+            var httpLimits = context.ServiceContext.ServerOptions.Limits;
+            var http3Limits = httpLimits.Http3;
+
+            _serverSettings.UpdateMaxHeaderListSize(httpLimits.MaxRequestHeadersTotalSize);
```

Comment:
I added these methods instead of having properties for tests to derive from Http3. This is because it's hard to know when settings are actually set in HTTP/3. In the test, I have a cts which sets the result when UpdateMaxHeaderListSize is called.

Reply:
Why are UpdateQPackBlockedStreams and UpdateQPackMaxTableCapacity also virtual methods if they're not overridden by tests? 

Also, the way we deal with stuff like this for HTTP/2 (there may be a single connection, but certain things would still normally get dispatched to the ThreadPool) is to configure an inline scheduler in the test base class. I think this is easier than adding a bunch of virtual methods.

Reply:
Yeah, I realize now this will always pass because Inline.